driver/makedepend.ml

```diff
@@ -646,6 +703,11 @@ let main () =
   Clflags.parse_arguments file_dependencies usage;
   Compenv.readenv ppf Before_link;
   if !sort_files then sort_files_by_dependencies !files
+  else if !print_json then begin
```

Comment:
I would prefer if the `print_json` conditional was close to the place in the code where we decide (or not) to use the `-modules` output. I believe this is within the `print_file_dependencies` function.

Reply:
@gasche 
would it be possible to achieve it without changing the current code structure of  `else` block ?
 `List.iter print_file_dependencies (List.sort compare !files);`
the `List.iter` is turning out to be a problem for Json printing, so does the code I pushed most look good ?